qa/test_framework/util.py

```diff
@@ -494,19 +495,31 @@ def get_coinbase_address(node, expected_utxos=None):
 """
 def initialize_new_sidechain_in_mainchain(mainchain_node, withdrawal_epoch_length, public_key, forward_transfer_amount,
                                           vrf_public_key, gen_sys_constant, cert_vk, csw_vk, btr_data_length,
-                                          sc_creation_version, is_csw_enabled):
+                                          sc_creation_version, is_csw_enabled, type_of_circuit_number):
     number_of_blocks_to_enable_sc_logic = 479
     number_of_blocks = mainchain_node.getblockcount()
     diff = number_of_blocks_to_enable_sc_logic - number_of_blocks
     if diff > 1:
         logging.info("Generating {} blocks for reaching needed mc fork point...".format(diff))
         mainchain_node.generate(diff)
 
+    if sc_creation_version <= 1:
+        assert type_of_circuit_number == 0
+    else:
+        assert type_of_circuit_number == 1
+
+    if type_of_circuit_number == 1:
+        assert is_csw_enabled is False
+
     custom_creation_data = vrf_public_key
 
-    fe_certificate_field_configs = certificate_field_config_csw_disabled
-    if is_csw_enabled:
-        fe_certificate_field_configs = certificate_field_config_csw_enabled
+    if type_of_circuit_number == 0:
+        fe_certificate_field_configs = certificate_field_config_csw_disabled
+
+        if is_csw_enabled:
+            fe_certificate_field_configs = certificate_field_config_csw_enabled
+    else:
+        fe_certificate_field_configs = certificate_with_key_rotation_field_config
```

Comment:
I propose to add more comments, explaining this tricky configs.
And moreover, we may restructure a bit:
```
if type_of_circuit_number == 0:
  assert sc_creation_version <= 1
  fe_certificate_field_configs = ce...

elif type_of_circuit_number == 1:
  assert sc_creation_version == 2
  assert is_csw_enabled is False
  fe_certificate_field_configs = certificate_with_key_rotation_field_config
  
```